### PR TITLE
Modify reporter discovery to look for references using DependencyModel

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/dotnet-test-xunit/Program.cs [hunk 1]
```diff
@@ -181,34 +182,32 @@ static void OnUnhandledException(object sender, UnhandledExceptionEventArgs e)
         static List<IRunnerReporter> GetAvailableRunnerReporters()
         {
             var result = new List<IRunnerReporter>();
-            var runnerPath = Path.GetDirectoryName(AppContext.BaseDirectory);
+            var dependencyModel = DependencyContext.Load(typeof(Program).GetTypeInfo().Assembly);
```
Comment (pranavkm) [inline src/dotnet-test-xunit/Program.cs:185]: This should actually be the the test project so that any reporters specified as dependencies in the test project light up here. Unfortunately the argument parsing is intractably linked to this piece of code running prior to it and I didn't have the time to refactor it.

**Comment by pranavkm** [comment]
cc @pakrym

File: dotnet-test-xunit.sln [hunk 2]
```diff
@@ -1,14 +1,12 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 14
-VisualStudioVersion = 14.0.23107.0
+VisualStudioVersion = 14.0.24720.0
 MinimumVisualStudioVersion = 14.0.22823.1
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{8EE7A674-CE00-403B-BBDE-1C250C7A1F52}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "test", "test", "{8B1FE5A0-EA59-456D-B293-82B806DDBEFD}"
 EndProject
```
Comment (pakrym) [inline dotnet-test-xunit.sln:9]: Why did you delete this project from solution?

File: src/dotnet-test-xunit/Program.cs [hunk 3]
```diff
@@ -18,6 +18,7 @@
 
 using ISourceInformationProvider = Xunit.Abstractions.ISourceInformationProvider;
 using VsTestCase = Microsoft.Extensions.Testing.Abstractions.Test;
+using Microsoft.Extensions.DependencyModel;
```
Comment (pakrym) [inline src/dotnet-test-xunit/Program.cs:21]: Sort.

File: src/dotnet-test-xunit/Program.cs [hunk 4]
```diff
@@ -181,34 +182,32 @@ static void OnUnhandledException(object sender, UnhandledExceptionEventArgs e)
...
                 catch
                 {
                     continue;
                 }
 
-                foreach (var type in types)
+                foreach (var type in assembly.DefinedTypes)
                 {
-                    if (type == null || type.IsAbstract || type == typeof(DefaultRunnerReporter).GetTypeInfo() || type.ImplementedInterfaces.All(i => i != typeof (IRunnerReporter)))
+                    if (type == null || type.IsAbstract || type == typeof(DefaultRunnerReporter).GetTypeInfo() || type.ImplementedInterfaces.All(i => i != typeof(IRunnerReporter)))
```
Comment (pakrym) [inline src/dotnet-test-xunit/Program.cs:203]: `IsAssignableFrom`?

**Comment by pakrym** [comment]
:speech_balloon: :arrow_right: :shipit:

File: src/dotnet-test-xunit/Program.cs (hunk 4 above)
Comment (pranavkm) [inline src/dotnet-test-xunit/Program.cs:203]: Ehh, this is a port of external code which would need to be merged someday. I'd rather not change it.

File: dotnet-test-xunit.sln (hunk 2 above)
Comment (pranavkm) [inline dotnet-test-xunit.sln:9]: I'll investigate. VS said it couldn't find it.

**Comment by pranavkm** [comment]
@livarcocc, we need this change in for test reporting to work in Teamcity. Could you have a look at it?

File: src/dotnet-test-xunit/Program.cs [hunk 5]
```diff
@@ -181,34 +182,32 @@ static void OnUnhandledException(object sender, UnhandledExceptionEventArgs e)
...
-
-                var dllName = Path.GetFileNameWithoutExtension(dllFile);
-
+                Assembly assembly;
                 try
                 {
-                    var assembly = Assembly.Load(new AssemblyName(dllName));
-                    types = assembly.DefinedTypes.ToArray();
+                    assembly = Assembly.Load(assemblyName.Name);
+
```
Comment (cesarblum) [inline src/dotnet-test-xunit/Program.cs:194]: nit: remove blank line

**Comment by pranavkm** [comment]
ping @livarcocc  - this is blocking us.

**Comment by livarcocc** [comment]
:shipit:

**Comment by pranavkm** [comment]
Thanks!
